server/src/main/java/org/elasticsearch/index/mapper/TypeFieldMapper.java

```diff
@@ -154,6 +159,33 @@ public Query termsQuery(List<?> values, QueryShardContext context) {
             }
         }
 
+        @Override
+        public Query rangeQuery(Object lowerTerm, Object upperTerm, boolean includeLower, boolean includeUpper, QueryShardContext context) {
+            if (hasDocValues()) {
+                return new TermRangeQuery(name(), lowerTerm == null ? null : indexedValueForSearch(lowerTerm),
+                        upperTerm == null ? null : indexedValueForSearch(upperTerm), includeLower, includeUpper);
+            } else {
+                // this means the index has a single type and the type field is implicit
+                DEPRECATION_LOGGER.deprecatedAndMaybeLog("range_single_type",
+                        "Running [range] query on [_type] field for an index with a single type. As types are deprecated, this functionality will be removed in future releases.");
+                String type = context.getMapperService().documentMapper().type();
+                Query result = new MatchAllDocsQuery();
+                if (lowerTerm != null && lowerTerm instanceof BytesRef) {
```

Comment:
we shouldn't be lenient in case `lowerTerm` doesnt't implement BytesRef

Reply:
I was wondering the same, do we want to throw an IAE in those cases?

Reply:
I would call `indexedValueForSearch`.